src/toil/jobStores/azureJobStore.py

```diff
@@ -84,87 +84,96 @@ def _fetchAzureAccountKey(accountName):
+    """
+    class Locator(namedtuple('Locator', ('accountName', 'accountKey', 'namePrefix',
+                                         'tableService', 'blobService')), AbstractJobStore.Locator):
+        """
+        Represents the location of a job store in Azure similar to how URLs are used to locate
+        resources on the Internet.  Has the following attributes:
+            * accountName: An Azure account name as a string.
+            * accountKey: The account key that corresponds with the account name as a string.
+            * namePrefix: The name prefix of the job store as a string.
+            * tableService: An Azure table service object.
+            * blobService: An Azure blob service object.
+
+        The syntax for an Azure job store locator string is as follows:
+            azure:<account>:<name prefix>
+
+        For name prefix syntax see exception messages in parse method below below.
+        """
+        jobStoreName = 'azure'
+
+        # Dots in container names should be avoided because container names are used in HTTPS bucket
+        # URLs where the may interfere with the certificate common name. We use a double
+        # underscore as a separator instead.
+        #
+        containerNameRe = re.compile(r'^[a-z0-9](-?[a-z0-9]+)+[a-z0-9]$')
+
+        # See https://msdn.microsoft.com/en-us/library/azure/dd135715.aspx
+        #
+        minContainerNameLen = 3
+        maxContainerNameLen = 63
+        maxNameLen = 10
+        nameSeparator = 'xx'  # Table names must be alphanumeric
+
+        @property
+        def jobStoreCls(self):
+            return AzureJobStore
+
+        @classmethod
+        def parse(cls, locator):
+            try:
+                account, prefix = locator.split(':')
+            except ValueError:
+                raise ValueError("The job store locator '%s' is invalid." % locator)
+            else:
+                if cls.nameSeparator in prefix:
+                    raise ValueError("Invalid name prefix '%s'. Name prefixes may not contain %s." %
+                                     (prefix, cls.nameSeparator))
+                elif not cls.containerNameRe.match(prefix):
+                    raise ValueError("Invalid name prefix '%s'. Name prefixes must contain only "
+                                     "digits, hyphens or lower-case letters and must not start or "
+                                     "end in a hyphen." % prefix)
+                elif len(prefix) > cls.maxContainerNameLen - cls.maxNameLen - len(cls.nameSeparator):
+                    raise ValueError(("Invalid name prefix '%s'. Name prefixes may not be longer "
+                                      "than 50 characters." % prefix))
+                else:
+                    key = _fetchAzureAccountKey(account)
+                    return cls(account, key, AzureJobStore._sanitizeTableName(prefix),
+                               TableService(account_key=key, account_name=account),
```

Comment:
Do you realize that constructing these is not within the responsibility of a Locator?